code/game/gamemodes/wizard/spellbook.dm

```diff
@@ -719,8 +719,12 @@
 		recoil(user)
 	else
 		user.mind.AddSpell(S)
-		user <<"<span class='notice'>you rapidly read through the arcane book. Suddenly you realize you understand [spellname]!</span>"
-		user.attack_log += text("\[[time_stamp()]\] <font color='orange'>[user.real_name] ([user.ckey]) learned the spell [spellname] ([S]).</font>")
+		user << "<span class='notice'>You rapidly read through \
+			the arcane book. Suddenly you realize you understand \
+			[spellname]!</span>"
+		user.attack_log += text("\[[time_stamp()]\] \
+			<font color='orange'>[user.real_name] ([user.ckey]) \

```

Comment:
`key_name()`; User attack logs don't really need to have the owner's name in them so you could just replace this with `has`.
<sub><sub><sub><sub>I wonder why orange</sub></sub></sub></sub>